apps/generator/lib/generator.js

```diff
@@ -867,34 +866,46 @@ class Generator {
   async generateFile(asyncapiDocument, fileName, baseDir) {
     const sourceFile = path.resolve(baseDir, fileName);
     const relativeSourceFile = path.relative(this.templateContentDir, sourceFile);
+    const relativeSourceDirectory = relativeSourceFile.split(path.sep)[0] || '.';
+  
     const targetFile = path.resolve(this.targetDir, this.maybeRenameSourceFile(relativeSourceFile));
     const relativeTargetFile = path.relative(this.targetDir, targetFile);
-
+    let shouldGenerate = true;
+  
     if (shouldIgnoreFile(relativeSourceFile)) return;
-
-    const shouldOverwriteFile = await this.shouldOverwriteFile(relativeTargetFile);
-    if (!shouldOverwriteFile) return;
-
+    
+    if (!(await this.shouldOverwriteFile(relativeTargetFile))) return;
+    let conditionalPath = '';
+    // It becomes deprecated with this PR, and soon will be removed.
+    // TODO: https://github.com/asyncapi/generator/issues/1553
     if (this.templateConfig.conditionalFiles?.[relativeSourceFile]) {
```

Comment:
```suggestion
    // conditionalFiles becomes deprecated with this PR, and soon will be removed.
    // TODO: https://github.com/asyncapi/generator/issues/1553
    if (this.templateConfig.conditionalFiles?.[relativeSourceFile]) {
```

it is good practice that while something is refactored or deprecated, to search through code the name of function/config/argument, in this case `conditionalFiles`

as you can see after searching, we have `validateConditionalFiles` in `apps/generator/lib/templateConfigValidator.js`

- mark it as deprecated
- make sure we have proper validation for the new conditionalGeneration

Reply:
Added comments where conditionalFiles are present added method for new conditionalGeneration thanks